### ISSUE TITLE
system.numbers is slow if use WHERE

### ISSUE THREAD

**Issue by ipozdnyakov** [ClickHouse/ClickHouse#56679]
These queries work pretty well:
```
SELECT number as week_num FROM system.numbers LIMIT 8
SELECT number as week_num FROM numbers(0,8)
```

But these, supposedly doing the same, do not complete in a reasonable amount of time:
```
SELECT number as week_num FROM system.numbers WHERE week_num < 8
SELECT number as week_num FROM system.numbers WHERE number BETWEEN 1 AND 8"
```

**How to reproduce**
* Which ClickHouse server version to use - 22.8.19.10
* Interface - DataGrip over port forwarding to server (or the same via Grafana)

**Comment by ipozdnyakov** [comment]
Duplicate https://github.com/ClickHouse/ClickHouse/issues/48582

**Comment by JackyWoo** [comment]
@ipozdnyakov fixed by #50909
